test/functional/sqlsrv/sqlsrv_ae_type_conversion_select.phpt

```diff
@@ -0,0 +1,310 @@
+--TEST--
+Test fetching data by conversion with CAST in the SELECT statement
+--SKIPIF--
+<?php require('skipif_versions_old.inc'); ?>
+--FILE--
+<?php
+require_once('MsCommon.inc');
+require_once('tools.inc');
+require_once('values.php');
+
+// Set up the columns and build the insert query. Each data type has an
+// AE-encrypted and a non-encrypted column side by side in the table.
+function FormulateSetupQuery($tableName, &$dataTypes, &$columns, &$insertQuery, $strsize, $strsize2)
+{
+    $columns = array();
+    $queryTypes = "(";
+    $valuesString = "VALUES (";
+    $numTypes = sizeof($dataTypes);
+
+    for ($i = 0; $i < $numTypes; ++$i) {
+        // Replace parentheses for column names
+        $colname = str_replace("($strsize)", "_$strsize", $dataTypes[$i]);
+        $colname = str_replace("($strsize2)", "_$strsize2", $colname);
+        $colname = str_replace("(max)", "_max", $colname);
+        $colname = str_replace("(5)", "_5", $colname);
+        $colname = str_replace("(36,4)", "_36_4", $colname);
+        $colname = str_replace("(32,4)", "_32_4", $colname);
+        $colname = str_replace("(28,4)", "_28_4", $colname);
+        $columns[] = new AE\ColumnMeta($dataTypes[$i], "c_".$colname."_AE"); // encrypted column
+        $columns[] = new AE\ColumnMeta($dataTypes[$i], "c_".$colname, null, true, true);// non-encrypted column
+        $queryTypes .= "c_"."$colname, ";
+        $queryTypes .= "c_"."$colname"."_AE, ";
```

Comment:
`$queryTypes` is a little confusing (sounds like it's getting the datatype of something). Perhaps rename is to `$queryColumns`? or `$queryFields`?